simulation/simple_sensor_simulator/include/simple_sensor_simulator/sensor_simulation/sensor_simulation.hpp

```diff
@@ -28,21 +29,75 @@ namespace simple_sensor_simulator
 class SensorSimulation
 {
 public:
-  void attachLidarSensor(
-    const double current_time, const simulation_api_schema::LidarConfiguration & configuration,
-    std::shared_ptr<rclcpp::Publisher<sensor_msgs::msg::PointCloud2>> publisher_ptr);
-  void attachDetectionSensor(
-    const double current_time,
-    const simulation_api_schema::DetectionSensorConfiguration & configuration,
-    std::shared_ptr<rclcpp::Publisher<autoware_perception_msgs::msg::DynamicObjectArray>>
-      publisher_ptr);
+  auto attachLidarSensor(
+    const double current_simulation_time,
+    const simulation_api_schema::LidarConfiguration & configuration, rclcpp::Node & node) -> void
+  {
+    if (configuration.architecture_type() == "tier4/proposal") {
```

Comment:
Please use enum value in protobuf.
See also https://github.com/tier4/scenario_simulator_v2/blob/247befcb17a413cd990c1843ead8152121d7a5e1/simulation/simulation_interface/proto/simulation_api_schema.proto#L196-L213.

Reply:
I don't understand the meaning of this review. Probably the code you are referencing is incorrect.